internal/http/check.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2024 Adevinta
+*/
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	report "github.com/adevinta/vulcan-report"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/adevinta/vulcan-check-sdk/agent"
+	"github.com/adevinta/vulcan-check-sdk/config"
+	"github.com/adevinta/vulcan-check-sdk/state"
+)
+
+// Check stores all the information needed to run a check locally.
+type Check struct {
+	Logger      *log.Entry
+	Name        string
+	checker     Checker
+	config      *config.Config
+	port        int
+	server      *http.Server
+	exitSignal  chan os.Signal // used to stopt the server either by an os Signal or by calling Shutdown()
+	shuttedDown chan int       // used to wait for the server to shut down.
+}
+
+// ServeHTTP implements an HTTP POST handler that receives a JSON encoded job, and returns an
+// agent.State JSON encoded response.
+func (c *Check) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would not call this function `ServeHttp`. It is like `Check` implements `http.Handler` but that's not the real purpose.

Reply:
I need it for the Shutdown method.

Reply:
I cannot see the change.

Reply:
Ok, I missed, but not sure what name is ok.